src/main/c/zlib/zlib.c

```diff
@@ -3408,7 +3408,15 @@ static VALUE
 rb_gzfile_total_out(VALUE obj)
 {
     struct gzfile *gz = get_gzfile(obj);
-    return rb_uint2inum(gz->z.stream.total_out - gz->z.buf_filled);
+    // fix from https://bugs.ruby-lang.org/issues/13616
```

Comment:
This seems fixed in Ruby 2.3.5, which version of zlib do we have?
We should have C-ext and test consistent to avoid headaches :)